### PR TITLE
CreationService - Issues #45 #46 #47

### LINKED ISSUE #45: Queue for adding an event through Creation service
## 🚀 Feature request
Implement an "**events**" queue which would subscribe to messages which have the "**events.add**"

### LINKED ISSUE #46: Get available cars through Creation service
## 🚀 Feature request
Create a request to the Connector for available cars using the "**connector.inbound**" queue with a "getAvailableCars" command

### LINKED ISSUE #47: Find drivers through Creation service
## 🚀 Feature request
Send a message to the "**driver.requests**" queue with enough data for it to find suitable drivers

### PR DESCRIPTION
**Pull request by davi7725**
## Checklist
- [x] Implement an "events" queue which would subscribe to messages which have the "events.add"
- [x] Create a request to the Connector for available cars using the "connector.requests" queue with a "getAvailableCars" command
- [x] Send a message to the "driver.find" queue with enough data for it to find suitable drivers

### PR Description
Created a new .Net Core console application.
Added functionality to subscribe to **event.add**, **cars.available** and **driver.found** queues.
Added functionality to publish to **connector.requests** and **driver.find** queues, on each message to these we send a "RequestId" which we use to correlate to an event using the _EventRequestRepository_.

### How to test
I have created a docker image of the creation service and it is pushed to docker hub

Start by executing the following docker commands:
- `docker network create testnet`

- `docker run --rm --name rabbitmq -d --network=testnet -p 15672:15672 -p 5672:5672 rabbitmq:management`
- `docker run --rm --name creationservice --network=testnet davi7816/testcreationservice`
(before succeeding to start the service, you should give some time to rabbitmq to start)

After all of these are started, you can test the functionality by heading to the rabbit MQ web interface at [http://localhost:15672/](http://localhost:15672/). In there head over to queues.

> Sending the request for a new event  

Publish the following message into the **event.add** queue.
`{ "command":"add", "EventId":"1", "Location":"Porto", "DriveFrom":"Lisbon", "DateFrom":"2015-06-23", "DateTo":"2015-06-25" } `
This will start the process of finding a car and a driver. You can verify this by publishing the next commands.

> Getting the needed RequestId identifier

You can get the needed identifier by heading to the following queues: **connector.requests** and **driver.find**, and getting a message. Each of this identifiers will be needed for the next messages.

> Responding with found cars  

Publish the following message into the **cars.available** queue. Dont forget to replace the _requestId_ with the one you obtained for the **connector.requests** queue in the previous step.
`{ "command":"carFound", "requestId":"6bb1e582-9592-44d7-9e0f-d82a7f737ce7", "cars": [{ "license":"123456789", "type":"A", "seats":4, "price":125 }] }  `

> Responding with found drivers

Publish the following message into the **driver.found** queue. Dont forget to replace the _requestId_ with the one you obtained for the **driver.find** queue in the previous step.
`{ "command":"driverFound", "requestId":"4c51322a-7d3d-456a-b093-31271511f3ff", "drivers": [{ "Name":"David" }] }  `

> Testing if everything went fine
If everything went fine you should now have a queue named **event.update**. Head over to this queue and get 2 messages, you should be able to get both of them, one with a "car" command and another with a "driver" command.

Issue reference

close #45 
close #46
close #47

### CODE REVIEW

File: CreationService/CommandRouter.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        DateTo = dateTo
+                    };
+
+                    EventRequest r = new EventRequest(Guid.NewGuid(), eventObj);
+                    EventRequestRepository.Instance.AddRequest(r);
+
+                    JObject requestCar = new JObject();
+                    requestCar["from"] = dateFrom;
+                    requestCar["to"] = dateTo;
+                    requestCar["stationId"] = driveFrom;
```
Comment (elit0451) [inline CreationService/CommandRouter.cs]: Wouldn't it be better to name it `station` as we are passing the station name?

File: CreationService/CommandRouter.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+                    MessageGateway.PublishMessage("connector.requests", requestCar.ToString());
+                    MessageGateway.PublishMessage("driver.find", requestDriver.ToString());
+                    break;
+
+                case "carFound":
+                    {
+                        string requestId = receivedObj["requestId"].Value<string>();
+                        JArray cars = receivedObj["cars"].Value<JArray>();
+                        string license = cars[0]["license"].Value<string>();
```
Comment (elit0451) [inline CreationService/CommandRouter.cs]: Here we always get the first element from the array?

Comment (davi7725) [reply CreationService/CommandRouter.cs]: Yes, we only care about taking whatever the first available car is.

File: CreationService/CommandRouter.cs (hunk 1 above)
Comment (davi7725) [reply CreationService/CommandRouter.cs]: The legacy system requires a string as an ID,l even though on the tests it is written "Porto", we might use something different such as "23".

**Comment by davi7725** [comment]
Thanks for the review!

**Pull request merged**